core/src/smartcontracts/isi/world.rs

```diff
@@ -150,6 +150,18 @@ pub mod isi {
                 state_transaction.world.accounts.remove(account);
             }
 
+            let remove_asset_definitions: Vec<AssetDefinitionId> = state_transaction
+                .world
+                .asset_definitions_in_domain_iter(&domain_id)
+                .map(|ad| ad.id().clone())
+                .collect();
+            for asset_definition_id in remove_asset_definitions {
+                state_transaction
+                    .world
+                    .asset_definitions
+                    .remove(asset_definition_id);
+            }
+
             if state_transaction
                 .world
                 .domains
```

Comment:
since this can fail, I think you should best do it first and only then clean the rest of the state